"register" keyword used in src/fft/plimpton/ files is deprecated and was removed in C++17

**Summary of issue**
<!-- Provide a succinct description of the issue in a few sentences-->

The keyword `register` appears in `src/fft/plimpton/` files 104 times. 

As noted [here](https://en.cppreference.com/w/cpp/keyword/register), the use of this keyword as a hint to the compiler to place the automatic storage duration variable in the processor registers was deprecated in C++11 and formally removed from the language in C++17. Since we are compiling with `--std=c++11`, by default, some compilers (later Intel versions) will throw warnings when building an Athena++ executable that was configured with `-fft`. As I have been messing around with some C++17 features via `--std=c++17`, I noticed that compilers will be more aggressive with warnings or even emit errors because of the presence of the `register` keyword:

```
src/fft/plimpton/pack_3d.cpp:234:60: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   register int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
```
Also note from [this StackOverflow post](https://stackoverflow.com/questions/20618008/replacement-for-deprecated-register-keyword-c-11):
> I don't think the "register" keyword ever affected register allocation in gcc. For that you have to go back to compilers of the 1970s.

**Additional comments and/or proposed solution**

Probably remove all 104 uses of the keyword?

--- a/src/fft/plimpton/pack_2d.cpp
+++ b/src/fft/plimpton/pack_2d.cpp
@@ -43,8 +43,8 @@
 void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 
 {
-  register int in,out,fast,slow;
-  register int nfast,nslow,nstride;
+  int in,out,fast,slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -64,8 +64,8 @@ void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register int in,out,fast,slow;
-  register int nfast,nslow,nstride;
+  int in,out,fast,slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -85,8 +85,8 @@ void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register int in,out,fast,slow;
-  register int nfast,nslow,nstride;
+  int in,out,fast,slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -106,8 +106,8 @@ void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register int in,out,fast,slow;
-  register int nfast,nslow,nstride;
+  int in,out,fast,slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -129,8 +129,8 @@ void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_n(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register int in,out,iqty,instart,fast,slow;
-  register int nfast,nslow,nstride,nqty;
+  int in,out,iqty,instart,fast,slow;
+  int nfast,nslow,nstride,nqty;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -161,9 +161,9 @@ void unpack_2d_permute_n(double *buf, double *data, struct pack_plan_2d *plan)
 void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -184,9 +184,9 @@ void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -207,9 +207,9 @@ void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -230,9 +230,9 @@ void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -255,9 +255,9 @@ void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_n(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,slow;
-  register int nfast,nslow,nstride,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,slow;
+  int nfast,nslow,nstride,nqty;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -291,9 +291,9 @@ void unpack_2d_permute_n(double *buf, double *data, struct pack_plan_2d *plan)
 void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out;
-  register int slow,size;
-  register int nfast,nslow,nstride;
+  double *in,*out;
+  int slow,size;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -313,9 +313,9 @@ void pack_2d(double *data, double *buf, struct pack_plan_2d *plan)
 void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out;
-  register int slow,size;
-  register int nfast,nslow,nstride;
+  double *in,*out;
+  int slow,size;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -335,9 +335,9 @@ void unpack_2d(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -358,9 +358,9 @@ void unpack_2d_permute_1(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int slow;
-  register int nfast,nslow,nstride;
+  double *in,*out,*begin,*end;
+  int slow;
+  int nfast,nslow,nstride;
 
   nfast = plan->nfast;
   nslow = plan->nslow;
@@ -383,9 +383,9 @@ void unpack_2d_permute_2(double *buf, double *data, struct pack_plan_2d *plan)
 void unpack_2d_permute_n(double *buf, double *data, struct pack_plan_2d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,slow;
-  register int nfast,nslow,nstride,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,slow;
+  int nfast,nslow,nstride,nqty;
 
   nfast = plan->nfast;
   nslow = plan->nslow;

--- a/src/fft/plimpton/pack_3d.cpp
+++ b/src/fft/plimpton/pack_3d.cpp
@@ -43,8 +43,8 @@
 void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -69,8 +69,8 @@ void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -95,8 +95,8 @@ void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -121,8 +121,8 @@ void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -149,8 +149,8 @@ void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,iqty,instart,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
+  int in,out,iqty,instart,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -178,8 +178,8 @@ void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -203,8 +203,8 @@ void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  int in,out,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -230,8 +230,8 @@ void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register int in,out,iqty,instart,fast,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
+  int in,out,iqty,instart,fast,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -266,9 +266,9 @@ void unpack_3d_permute2_n(double *buf, double *data, struct pack_plan_3d *plan)
 void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -294,9 +294,9 @@ void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -322,9 +322,9 @@ void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -350,9 +350,9 @@ void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -380,9 +380,9 @@ void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -411,9 +411,9 @@ void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -438,9 +438,9 @@ void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -467,9 +467,9 @@ void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -507,9 +507,9 @@ void unpack_3d_permute2_n(double *buf, double *data, struct pack_plan_3d *plan)
 void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out;
-  register int mid,slow,size;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane,upto;
+  double *in,*out;
+  int mid,slow,size;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane,upto;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -535,9 +535,9 @@ void pack_3d(double *data, double *buf, struct pack_plan_3d *plan)
 void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out;
-  register int mid,slow,size;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane,upto;
+  double *in,*out;
+  int mid,slow,size;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane,upto;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -563,9 +563,9 @@ void unpack_3d(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -591,9 +591,9 @@ void unpack_3d_permute1_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -621,9 +621,9 @@ void unpack_3d_permute1_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -652,9 +652,9 @@ void unpack_3d_permute1_n(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -679,9 +679,9 @@ void unpack_3d_permute2_1(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*begin,*end;
-  register int mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane;
+  double *in,*out,*begin,*end;
+  int mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane;
 
   nfast = plan->nfast;
   nmid = plan->nmid;
@@ -708,9 +708,9 @@ void unpack_3d_permute2_2(double *buf, double *data, struct pack_plan_3d *plan)
 void unpack_3d_permute2_n(double *buf, double *data, struct pack_plan_3d *plan)
 
 {
-  register double *in,*out,*instart,*begin,*end;
-  register int iqty,mid,slow;
-  register int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
+  double *in,*out,*instart,*begin,*end;
+  int iqty,mid,slow;
+  int nfast,nmid,nslow,nstride_line,nstride_plane,nqty;
 
   nfast = plan->nfast;
   nmid = plan->nmid;

